lib/anyway/config.rb

```diff
@@ -126,14 +128,36 @@ def config_attributes
         end
       end
 
-      def required(*names)
-        unless (unknown_names = (names - config_attributes)).empty?
-          raise ArgumentError, "Unknown config param: #{unknown_names.join(",")}"
-        end
+      def required(*names, env: nil)
+        unknown_names = names - config_attributes
+        raise ArgumentError, "Unknown config param: #{unknown_names.join(",")}" if unknown_names.any?
 
+        names = filter_by_env(names, env)
         required_attributes.push(*names)
       end
 
+      def filter_by_env(names, env)
+        return names if ["", current_env].any?(env.to_s)
```

Comment:
Let's avoid this style of allocating an array where simple `&&` works (and looks more readable)

```suggestion
        return names if current_env.empty? || current_env == env.to_s
```